### PR TITLE
Define and activate KVM 12.0.0 release with Kubernetes 1.17

### PR DESCRIPTION
**Pull request by tfussell**

### CODE REVIEW

**Comment by sslavic** [comment]
We successfully tested most of the apps https://github.com/giantswarm/giantswarm/issues/11154#issuecomment-645542121

Thanks for preparing the release!

**Comment by sslavic** [comment]
@MarcelMue @tfussell what's state of this one, and current plan for shipping it? 

To finish shipping app updates (it seems only nginx applied for kvm), need:
- either next patch 11.3.3
- join 12.0.0
- ship as followup patch 12.0.1

WDYT?

**Comment by sslavic** [comment]
Created PR to update apps in KVM 12.0.0 https://github.com/giantswarm/releases/pull/335, please have a look.

File: kvm/v12.0.0/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+- **CSI Migration Beta**: The Kubernetes in-tree storage plugin to Container Storage Interface (CSI) migration infrastructure is now beta in Kubernetes v1.17. CSI migration was introduced as alpha in Kubernetes v1.14.
+
+### kvm-operator v3.11.2
+- Modified Calico deployment to use `-bird-live` as a liveness probe improving observability of failed mesh networking.
+- Updated node setup for 1.17 compatibility.
+- Removed limits from `calico-kube-controllers`.
+
+### Calico v3.14.0
+- Updated from v3.10.1 - [changelog](https://docs.projectcalico.org/v3.14/release-notes/).
+- TODO
```
Comment (MartaD) [inline kvm/v12.0.0/README.md:18]: Fixes IPv6 rogue router advertisement vulnerability
  [CVE-2020-13597](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-13597).

File: kvm/v12.0.0/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+### Kubernetes v1.17.6
+- Updated from v1.16.9 - [changelog](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.17.md#changelog-since-v1175).
+- **Cloud Provider Labels reach General Availability**: Added as a beta feature way back in v1.2, v1.17 sees the general availability of cloud provider labels.
+- **Volume Snapshot Moves to Beta**: The Kubernetes Volume Snapshot feature is now beta in Kubernetes v1.17. It was introduced as alpha in Kubernetes v1.12, with a second alpha with breaking changes in Kubernetes v1.13.
+- **CSI Migration Beta**: The Kubernetes in-tree storage plugin to Container Storage Interface (CSI) migration infrastructure is now beta in Kubernetes v1.17. CSI migration was introduced as alpha in Kubernetes v1.14.
+
+### kvm-operator v3.11.2
+- Modified Calico deployment to use `-bird-live` as a liveness probe improving observability of failed mesh networking.
+- Updated node setup for 1.17 compatibility.
+- Removed limits from `calico-kube-controllers`.
```
Comment (MartaD) [inline kvm/v12.0.0/README.md:14]: Can you explain why those limits were removed?

Comment (tfussell) [reply kvm/v12.0.0/README.md:14]: We decided towards the end of last year to drop CPU and memory limits for all core components to ensure that they are not OOM killed. We replaced the limits with alerts about high memory usage which we can investigate and act on manually. Here's the issue for this specific change https://github.com/giantswarm/giantswarm/issues/10543

Comment (MartaD) [reply kvm/v12.0.0/README.md:16]:
```suggestion
- Removed limits from `calico-kube-controllers` to prevent it from being OOM killed.
```
How about this?

**Comment by sslavic** [comment]
Tested on geckon:
- 12.0.0 cluster creation
- upgrade from 11.3.2 to 12.0.0

Reproduced known issue https://github.com/giantswarm/giantswarm/issues/11583
Should we include the fix in 12.0.0 or follow up patch release? Is it enough to revert https://github.com/giantswarm/k8s-network-bridge/pull/38, update where it's used?

**Comment by calvix** [comment]
rather than reverting I would just remove the lines
```
https://github.com/giantswarm/k8s-network-bridge/pull/38/files#diff-60713fb19f6c50a2473835df63ff17f9R179-R184
```

File: kvm/v12.0.0/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+## :zap: Giant Swarm Release 12.0.0 for KVM :zap:
+
+This is the first Giant Swarm release which includes Kubernetes v1.17. Components have been updated for improved reliability and observability. Further details about changes to individual components can be found below.
+
+### Kubernetes v1.17.6
```
Comment (stone-z) [inline kvm/v12.0.0/README.md:5]:
```suggestion
### Kubernetes v1.17.8
```

File: kvm/v12.0.0/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+This is the first Giant Swarm release which includes Kubernetes v1.17. Components have been updated for improved reliability and observability. Further details about changes to individual components can be found below.
+
+### Kubernetes v1.17.8
+
+- Updated from v1.16.9 - [changelog](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.17.md#changelog-since-v1177).
+- **Cloud Provider Labels reach General Availability**: Added as a beta feature way back in v1.2, v1.17 sees the general availability of cloud provider labels.
+- **Volume Snapshot Moves to Beta**: The Kubernetes Volume Snapshot feature is now beta in Kubernetes v1.17. It was introduced as alpha in Kubernetes v1.12, with a second alpha with breaking changes in Kubernetes v1.13.
+- **CSI Migration Beta**: The Kubernetes in-tree storage plugin to Container Storage Interface (CSI) migration infrastructure is now beta in Kubernetes v1.17. CSI migration was introduced as alpha in Kubernetes v1.14.
+
```
Comment (stone-z) [inline kvm/v12.0.0/README.md:11]: Flatcar update might be worth mentioning somewhere as well

**Comment by tfussell** [comment]
Generated by `devctl` with:
```
devctl create release --name 12.0.0 \
  --provider kvm \
  --base 11.3.2 \
  --app cert-exporter@1.2.3 \
  --app coredns@1.1.10@1.6.5 \
  --app kube-state-metrics@1.1.0@1.9.5 \
  --app metrics-server@1.1.0@0.3.3 \
  --app net-exporter@1.9.0 \
  --app nginx-ingress-controller@1.7.1@0.33.0 \
  --app node-exporter@1.2.0@0.18.1 \
  --component calico@3.14.1 \
  --component etcd@3.4.9 \
  --component kubernetes@1.17.8 \
  --component kvm-operator@3.12.0 \
  --component cluster-operator@0.23.10 \
  --component containerlinux@2512.2.1 \
  --overwrite
```

File: kvm/v12.0.0/README.md (hunk 4 above)
Comment (tfussell) [reply kvm/v12.0.0/README.md:29]: For sure. It's there now.

File: kvm/v12.0.0/release.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    version: 2512.2.1
+  - name: etcd
+    version: 3.4.9
+  - name: flannel-operator
+    version: 0.2.0
+  - name: kubernetes
+    version: 1.17.8
+  - name: kvm-operator
+    version: 3.12.0
+  date: "2020-06-11T15:00:00Z"
```
Comment (MarcelMue) [inline kvm/v12.0.0/release.yaml:48]: Update this before merging.

File: kvm/v12.0.0/announcement.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# :zap: Giant Swarm Release 12.0.0 for KVM :zap:
+
+KVM release 12.0.0 is now available. This is the first Giant Swarm release which includes Kubernetes v1.17. Many core components and default apps have been updated for improved reliability and observability. Further details about changes to individual components can be found by visiting [this link](https://github.com/giantswarm/releases/blob/master/kvm/v12.0.0).
```
Comment (MartaD) [inline kvm/v12.0.0/announcement.md:3]: As per Slack discussion, wdyt @tfussell ?
```suggestion
KVM release 12.0.0 is now available. This is the first Giant Swarm release which includes Kubernetes v1.17. Many core components and default apps have been updated for improved reliability and observability. Further details about changes to individual components can be found by visiting [this link](https://github.com/giantswarm/releases/blob/master/kvm/v12.0.0).

Please reply to this announcement in a thread if you have any feedback or questions.
<!subteam^S0GSG846L> 
```

Comment (MartaD) [reply kvm/v12.0.0/announcement.md:3]: or:
Please help <!subteam^S0GSG846L> improve communication and reply to this announcement in a thread if you have any feedback or questions.

wierd string will be replaced with ProductOwners

**Comment by MartaD** [comment]
ping: sig-product for full release notes review.

**Comment by MarcelMue** [comment]
IMO we should merge here.

**Pull request merged**